R/utils.R

```diff
@@ -53,11 +53,31 @@ parse_size <- function(x) {
   )
 }
 
-id_field <- function(id, field, default = NULL) {
-  if (field %in% names(id@name)) {
-    id@name[[field]]
+id_field <- function(id,
+                     field = c("catalog", "schema", "table"),
+                     error_call = caller_env()) {
+  arg_match(field, error_call = error_call)
+
+  if (length(id@name) == 1) {
```

Comment:
This is the main fix; we can no longer assume that the `Id()` object is named, so we instead rely on position.